pytest_base_url/plugin.py

```diff
@@ -21,13 +24,21 @@ def base_url(request):
 def _verify_url(request, base_url):
     """Verifies the base URL"""
     verify = request.config.option.verify_base_url
+    verify_timeout = request.config.getoption('verify_base_url_attempts')
     if base_url and verify:
-        response = requests.get(base_url, timeout=10)
-        if not response.status_code == requests.codes.ok:
-            raise pytest.UsageError(
-                'Base URL failed verification!'
-                '\nURL: {0}, Response status code: {1.status_code}'
-                '\nResponse headers: {1.headers}'.format(base_url, response))
+        session = requests.Session()
+        retries = Retry(total=verify_timeout,
+                        backoff_factor=0.1,
+                        status_forcelist=[500, 502, 503, 504])
+        session.mount(base_url, HTTPAdapter(max_retries=retries))
+        try:
+            session.get(base_url)
+        except Exception:
```

Comment:
The exception is a ```ConnectionError``` not a specific python one however. The output from just failing to get the url is: 
```
requests.exceptions.ConnectionError: HTTPConnectionPool(host='request.get', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x10310cf60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
```

From this one could inference that the URL didn't load, but its not helpfulas to why.